server/odc-service/src/main/java/com/oceanbase/odc/metadb/dlm/TaskUnitRepository.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.metadb.dlm;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import com.oceanbase.odc.config.jpa.OdcJpaRepository;
+
+/**
+ * @Author：tinker
+ * @Date: 2023/8/10 15:20
+ * @Descripition:
+ */
+public interface TaskUnitRepository extends OdcJpaRepository<TaskUnitEntity, Long> {
+
+    List<TaskUnitEntity> findByGeneratorId(@Param("generatorId") String generatorId);
+
+    Optional<TaskUnitEntity> findByJobIdAndGeneratorIdAndTaskIndex(String jobId, String generatorId, Long taskIndex);
+
+    @Query(value = "select count(1) from dlm_task_unit where job_id=:jobId and (status != 'SUCCESS' or primary_key_cursor is null)",
```

Comment:
is `primary_key_cursor is null` a valid grammar in h2?

Reply:
DLM is unsupported in client. We will not modify it this time and will reconstruct it later.